### ISSUE TITLE
`fleek add` doesn't

### ISSUE THREAD

**Issue by mtalexan** [ublue-os/fleek#237]
**Describe the bug**  
`fleek apply` doesn't install any of the `fleek add`ed packages or make any of the custom aliases available.

**To Reproduce**  
1. Install nix on a fresh system with the Determinate Nix installer
2. Install fleek with the curl command
3. `fleek init`
4. `fleek add neovim`
7. `fleek apply -v`
8. Open new terminal
9. `fleeks`
10. `alias`
11. `which nvim`
12. `nix profile list`
13. `cat ~/.fleek.yaml`

**Actual behavior** 
`fleeks` reports the command is unknown.  
`alias` reports the `fleeks` alias doesn't exist.  
`which nvim` reports no such command is found.  
`nix profile list` reports that only `home-manager-path` is installed and not `neovim`.  

**Expected behavior**  
The default `fleeks` alias should exist and work.  
The `nvim` command, which is installed as part of `neovim` should be installed in the nix profile.  

**Environment**
 - OS: Fedora 38 (fresh install)
 - Shell: bash
 - fleek version 0.9.16

**Additional context**

The `~/.fleek.yaml` lists `neovim` in the `packages` section after the `fleek add neovim`, and the `aliases` section does include the default `fleeks` alias.

---

EDIT: It turns out the issue is `fleek add` adding to the wrong list in the `.fleek.yaml` so the package isn't usable (not sure what the `package` list can even be used for, it seems to be completely non-functional).

**Comment by mtalexan** [comment]
EDIT: So apparently I was looking at the index in `nix profile list` with my initial report, not the home-manager generation, so I've updated the description with the more specific usage problem I'm encountering: nothing actually got installed.

**Comment by mtalexan** [comment]
Maybe fleek is only adding the home-manager config for these packages and isn't installing the actual nix packages?  If so, that would probably be at least part of the bug, but doesn't explain the aliases not working.  

I can run `nix profile install nixpkgs#neovim` and it installs it successfully.

**Comment by mtalexan** [comment]
Confirmed.  `fleek add` adds to the `packages` list and not the `programs` list.  Also, having the same package in both the `packages` and `programs` list causes a conflict.  
So `fleek add` is not operating like `nix profile add`, it instead can only (possibly) be used if you manually installed a nix program with `nix profile install`, and now want to add configuration for the home-manager (I assume).

**Comment by mtalexan** [comment]
At this point there's no documentation on what was intended from these, so I'm just guessing.

In the current state it appears `fleek` can be used to (mutually exclusively) define the nix packages you want installed (`programs` section), or define the home-manager config of a package you installed manually (`packages` section).  Fleek produces an error if the same package is listed in both places, even though that seems to be the exact use case for the tool.

**Comment by mtalexan** [comment]
I just cleared out the old fleek config and re-installed:  
1. `fleek eject`
2. `rm ~/.fleek.yaml`
3. `rm -rf ~/.local/share/fleek`

And re-installed it from scratch with git already setup:  
1. `curl -fsSL https://getfleek.dev/installer | bash`
    1. Accept detected `git config --global user.email`
    2. Pick previously setup `GitHub` SSH key

Then edit the `.fleek.yaml` to set `bling: none`.  Then I ran the all-in-one command `fleek add --apply neovim` and got a warning message about my `.bash_profile` and `.bashrc` being in the way and getting backed up (for the first time ever).  

The `nix profile list` still only lists the `home-manager-path`, but `which nvim` is successfully finding the package now.

---

So it seems there's a bug somewhere in the `fleek` installation and `fleek init` process. Likely an interrupted `fleek init` (which is guaranteed to happen since the GitHub email and SSH key requirements are undocumented) can cause a screwed up init state that in my case manifested as home-manager not actually managing the environment.
